src/pages/Script/ScriptsComp.tsx

```diff
@@ -26,29 +26,37 @@ import { useSetToast } from '../../components/Toast'
 export const ScriptTransactions = ({ page, size }: { page: number; size: number }) => {
   const history = useHistory()
   const { codeHash, hashType } = useParams<{ codeHash: string; hashType: string }>()
+  const [transactionsEmpty, setTransactionsEmpty] = useState(false)
 
-  const transactionsQuery = useQuery(['scripts_ckb_transactions', codeHash, hashType, page, size], async () => {
-    const { data, meta } = await explorerService.api.requesterV2
-      .get(`scripts/ckb_transactions`, {
-        params: {
-          code_hash: codeHash,
-          hash_type: hashType,
-          page,
-          page_size: size,
-        },
-      })
-      .then((res: AxiosResponse) =>
-        toCamelcase<Response.Response<{ ckbTransactions: CkbTransactionInScript[] }>>(res.data),
-      )
+  const transactionsQuery = useQuery(
+    ['scripts_ckb_transactions', codeHash, hashType, page, size],
+    async () => {
+      const { data, meta } = await explorerService.api.requesterV2
+        .get(`scripts/ckb_transactions`, {
+          params: {
+            code_hash: codeHash,
+            hash_type: hashType,
+            page,
+            page_size: size,
+          },
+        })
+        .then((res: AxiosResponse) =>
+          toCamelcase<Response.Response<{ ckbTransactions: CkbTransactionInScript[] }>>(res.data),
+        )
 
-    if (data == null || data.ckbTransactions == null || data.ckbTransactions.length === 0) {
-      throw new Error('Transactions empty')
-    }
-    return {
-      total: meta?.total ?? 0,
-      ckbTransactions: data.ckbTransactions,
-    }
-  })
+      if (data == null || data.ckbTransactions == null || data.ckbTransactions.length === 0) {
```

Comment:
`data.ckbTransactions.length === 0` should not be included to disable `refetch` because an empty array of a transaction list is valid. The refetch can be disabled when API implies the query is invalid, e.g. query parameter is invalid.

Imagine a case that, when a script is just deployed, it should have no related transactions, but update of the list should still be automatic

Reply:
`==` should be avoided and `===` should be used